src/sentry/incidents/endpoints/organization_alert_rule_available_action_index.py

```diff
@@ -27,33 +31,37 @@ def fetch_pagerduty_services(self, organization, integration_id):
 
     def build_action_response(self, organization, registered_type, integration=None):
         allowed_target_types = [
-            action_target_type_to_string[target_type]
+            action_target_type_to_string.get(target_type)
             for target_type in registered_type.supported_target_types
         ]
 
-        input_type = (
-            "select"
-            if registered_type.type.value
-            in [
-                AlertRuleTriggerAction.Type.PAGERDUTY.value,
-                AlertRuleTriggerAction.Type.EMAIL.value,
-            ]
-            else "text"
-        )
-
         action_response = {
             "type": registered_type.slug,
             "allowedTargetTypes": allowed_target_types,
-            "integrationName": integration.name if integration else None,
-            "integrationId": integration.id if integration else None,
-            "inputType": input_type,
         }
 
-        if (
-            integration
-            and registered_type.type.value == AlertRuleTriggerAction.Type.PAGERDUTY.value
-        ):
-            action_response["options"] = self.fetch_pagerduty_services(organization, integration.id)
+        if registered_type.type.value in [
+            AlertRuleTriggerAction.Type.PAGERDUTY.value,
+            AlertRuleTriggerAction.Type.EMAIL.value,
+        ]:
+            action_response["inputType"] = "select"
+        elif registered_type.type.value in [
+            AlertRuleTriggerAction.Type.SLACK.value,
+            AlertRuleTriggerAction.Type.MSTEAMS.value,
+        ]:
+            action_response["inputType"] = "text"
+
+        if integration:
+            action_response["integrationName"] = integration.name
+            action_response["integrationId"] = integration.id
```

Comment:
> Are we setting `integrationId` in the case it's a Sentry App type?

So I had talked to @mgaeta about this outside of the comments I made but if we go this route, we have to make sure to update the `CamelSnakeModelSerializer` because that will automatically convert the integrationId to the model field `integration` which we don't want. This will reduce the frontend logic a lot, but it does feel a bit hacky.

Reply:
I've updated the API to pass around a separate `sentryAppId` instead of shoving it in. The code's much cleaner and I think it'll solve some problems down the line with the serializer.